nwb_conversion_tools/datainterfaces/moviedatainterface.py

```diff
@@ -0,0 +1,175 @@
+            reduced conversion speed (compared to loading video entirely into RAM as an array). This will also force to
+            True, even if manually set to False, whenever the video file size exceeds available system RAM by a factor
+            of 70 (from compression experiments). Based on experiements for a ~30 FPS system of ~400 x ~600 color
+            frames, the equivalent uncompressed RAM usage is around 2GB per minute of video. The default is True.
+        """
+        file_paths = self.source_data['file_paths']
+
+        if stub_test:
+            count_max = 10
+        else:
+            count_max = np.inf
+        if starting_times is not None:
+            assert isinstance(starting_times, list) and all([isinstance(x, float) for x in starting_times]) \
+                and len(starting_times) == len(file_paths), \
+                "Argument 'starting_times' must be a list of floats in one-to-one correspondence with 'file_paths'!"
+        else:
+            starting_times = [0.]
+
+        for j, file in enumerate(file_paths):
+            timestamps = starting_times[j] + get_movie_timestamps(movie_file=file)
+
+            if len(starting_times) != len(file_paths):
+                starting_times.append(timestamps[-1])
+
+            image_series_kwargs = dict(
+                name=f"Video: {Path(file).stem}",
+                description="Video recorded by camera.",
+                unit="Frames"
+            )
+            if check_regular_timestamps(ts=timestamps):
+                fps = get_movie_fps(movie_file=file)
+                image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
+            else:
+                image_series_kwargs.update(timestamps=H5DataIO(timestamps, compression="gzip"))
+
+            uncompressed_estimate = Path(file).stat().st_size * 70
+            available_memory = psutil.virtual_memory().available
+            if not chunk_data and uncompressed_estimate >= available_memory:
+                warn(
+                    f"Not enough memory (estimated {round(uncompressed_estimate/1e9, 2)} GB) to load movie file as "
+                    f"array ({round(available_memory/1e9, 2)} GB available)! Forcing chunk_data to True."
+                )
+                chunk_data = True
+
+            total_frames = len(timestamps)
+            frame_shape = get_frame_shape(movie_file=file)
+            maxshape = [total_frames]
+            maxshape.extend(frame_shape)
+            best_gzip_chunk = (1, frame_shape[0], frame_shape[1], 3)
+            tqdm_pos, tqdm_mininterval = (0, 10)
+            if chunk_data:
+                def data_generator(file, count_max):
+                    cap = cv2.VideoCapture(str(file))
+                    n_frames = 0
+                    success, frame = cap.read()
+                    # with tqdm(total=total_frames, desc="Writing movie data", position=0) as pbar:
+                    while success and n_frames < count_max:
```

Comment:
Can't you use a for loop knowing the number of frames?

Reply:
Hmm, a fair point since I do extract the total_frames (to get the timestamps outside a data generator) prior to those later data copies, so yes that could be simplified. I'll work on that right now